test_apps/smokeTests/gridTools.cpp

```diff
@@ -314,6 +315,31 @@ void PrintStats(double rms, size_t numMissingValues, size_t disagreements, doubl
     cout << endl;
 }
 
+// Copy of Grid::GetRange() function from VAPOR release 3.6
+//
+void GetRange_36(VAPoR::Grid* g, float range[2])
```

Comment:
@shaomeng maybe we should just leave the original serial implementations inside of Grid.cpp and call it something like GetRangeSerial? I'm not sure it if is a good model that every time we parallelize a portion of the code we move the serial version into the test driver. What do you think?

Reply:
I think for complex functions, the approach you described makes sense. But finding range is something so simple that I don't think there's much value to bloat Grid with two versions of it. 

That said, I can move it back if you think it's necessary.

Reply:
I don't think it is necessary for  this function. Just thinking about how we deal with the general issue of verifying that  parallel versions of a function produce correct results. One way to do this is to compare with the serial implementation (assuming we accept the serial code as ground truth). I don't think copying, for example, the flow integrator into this file is something we'll want to do...